dhis-2/dhis-api/src/main/java/org/hisp/dhis/dashboard/Dashboard.java

```diff
@@ -52,6 +53,16 @@
     
     private List<DashboardItem> items = new ArrayList<>();
 
+    /**
+     * Whether we restrict the filter dimensions the user may select.
+     */
+    private Boolean filterDimensionsAreRestricted;
+
+    /**
+     * Allowed filter dimensions (if any) the user may select.
+     */
+    private Set<String> allowedFilterDimensions;
```

Comment:
Instantiate set as HashSet to avoid nullpointer exceptions.

Reply:
I have now done that. At first I didn't do so because I didn't want there to be an empty Set by default -- which would mean disallowing all filter dimensions. But even after I added the boolean I didn't see a need for initializing an empty Set here. In my testing without initializing the field, it was always loaded either from the database or from json through the WebAPI and it never hit an error. I didn't include the initialization to a new HashSet because it seemed to be an unnecessary call, and I don't like to write unnecessary code.

But I know that initializing to a new HashSet is present in other such json Set fields. I have now initialized it so that (A) it is like the others, (B) we won't get a nullpointer should we try to reference the Set in the backend in other ways in the future, (C) in case there is some present circumstance I am not aware of where it would hit a nullpointer exception, and (D) you asked me to. ;)